pagy.gemspec

```diff
@@ -23,6 +23,7 @@ Gem::Specification.new do |s|
   s.add_development_dependency 'bundler',  '~> 1.16'
   s.add_development_dependency 'rake',     '~> 10.0'
   s.add_development_dependency 'minitest', '~> 5.0'
+  s.add_development_dependency 'rack'
```

Comment:
I don’t know if `rack` should in fact be considered a runtime dependency? You cannot use `Pagy::Frontend` without either having `Rack` loaded, or overriding `Pagy::Frontend#pagy_url_for`.

Reply:
> You cannot use Pagy::Frontend without either having Rack loaded, or overriding Pagy::Frontend#pagy_url_for

True, but...

The `pagy_url_for` uses rack just to give a default usable on most apps out of the box. Adding an explicit dependency would cause the installation of rack even if you never intended to use rack in your app and you are fine with just overriding the `pagy_url_for`.

On the other hand not adding it would cause the method to fail until the user will provide his/her own way to create the url, with no feedback added.

The best (puristic) approach would probably be conditionally defining the method in presence of rack, and defining the method raising a `NotImplementedError` exception if Rack is not present.

Something similar to how the `pagy_t` is implemented.

Reply:
Yeah, that sounds like a valid strategy. Except, just use `NoMethodError` instead of `NotImplementedError`: http://chrisstump.online/2016/03/23/stop-abusing-notimplementederror/ (as in: just leave the method not implemented, or raise it with a message about using Rack)

Reply:
I am fine with both options. Whatever is better for you. Thanks.

Reply:
I agree. I have removed the last commit, so this PR is back at just using `request.path` and adding a few tests where `Rack` needs to be a development dependency.

Reply:
Thanks